app/src/composables/useWalletChecks.ts

```diff
@@ -0,0 +1,100 @@
+import { ref, watch } from 'vue'
+import { useToastStore } from '@/stores'
+import {
+  useConnect,
+  useSwitchChain,
+  useAccount,
+  type Connector,
+  type CreateConnectorFn
+} from '@wagmi/vue'
+import { NETWORK } from '@/constant'
+import { log, parseError } from '@/utils'
+
+export function useWalletChecks() {
+  const isProcessing = ref(false)
+  const isSuccess = ref(false)
+  const isConnectRequired = ref(false)
+
+  const { addErrorToast } = useToastStore()
+  const { connectors, connect, error: connectError, isPending: isPendingConnect } = useConnect()
+  const { switchChain } = useSwitchChain()
+  const { isConnected } = useAccount()
+
+  watch(connectError, (newConnectError) => {
+    if (newConnectError) {
+      addErrorToast('Failed to connect wallet')
+      log.error('connectError.value', newConnectError)
+      resetRefs()
+    }
+  })
+
+  watch(isPendingConnect, (newStatus) => {
+    if (!newStatus && isConnected.value) {
+      isSuccess.value = true
+    }
+  })
+
+  function resetRefs() {
+    isConnectRequired.value = false
+    isSuccess.value = false
+    isProcessing.value = false
+  }
+
```

Comment:
I don't understand why you need this function.
You should create computed property and it will be better.